### PR TITLE
Fix #75

### LINKED ISSUE #75: Bug: concurrent subscription failure
I am receiving errors in SubAsync like:
`Can not subscribe observer. Each observer can only be subscribed once. (Parameter 'observer')` 
`Could not create subscription. Id='0HM77NV3SLV1L'. Subject='qmp.gi.292c4c4e-fdd6-4512-b7ee-fea6f3f523b7.>' QueueGroup=''.`

In our app subscriptions are created concurrently and from time to time they get the same id, which leads to these errors. I found out that the reason is how method `UniqueId.Generate` is implemented. The method is based on `DateTime.UtcNow.Ticks` which can lead to the same id in concurrent calls. I think this can be fixed by using an incremental value or something similar.

### PR DESCRIPTION
**Pull request by igoree**
Closes #75

### CODE REVIEW

**Review by danielwertheim — COMMENTED** [review]
Thanks for the PR. Left two comments. If you have time, it would be awesome if you could look into those. Thanks

File: src/main/MyNatsClient/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿using System.Runtime.CompilerServices;
```
[the remark this answers is not in the document]
Comment (igoree) [reply src/main/MyNatsClient/AssemblyInfo.cs:1]: Ok

File: src/main/MyNatsClient/Internals/UniqueId.cs [hunk 2]
```diff
@@ -11,9 +12,10 @@ internal static class UniqueId
             'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T',
             'U', 'V'
         };
+        private static long _idBase = DateTime.UtcNow.Ticks;
 
         internal static string Generate()
-            => string.Create(13, DateTime.UtcNow.Ticks, (buffer, val) =>
+            => string.Create(13, Interlocked.Increment(ref _idBase), (buffer, val) =>
```
[the remark this answers is not in the document]
Comment (igoree) [reply src/main/MyNatsClient/Internals/UniqueId.cs:18]: GUID is the best choice for this. I thought you had a reason not to use a GUID so I've just fixed your code.

**Comment by igoree** [comment]
I've replaced the id generation code with `Guid.NewGuid().ToString('N')`. We can change the type of subscription id to Guid but this is a breaking change. However, Guid will be slightly more performant in comparisons.

File: src/main/MyNatsClient/Internals/UniqueId.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielwertheim) [reply src/main/MyNatsClient/Internals/UniqueId.cs]: There was a reason. As it needs to be converted to a string in the end https://github.com/danielwertheim/mynatsclient/blob/master/src/main/MyNatsClient/NatsClient.cs#L60

The intention was to take a first step closer to less memory footprint when generating etc. But this is OK.

**Pull request merged**